### DISCUSSION TITLE
Asynchronous scoped sessions not torn down and running in a different loop blocking main thread.

### DISCUSSION THREAD

**Discussion by EM51641** [pallets/flask#5265] [Q&A]
Hello,

First, I want to thank you for this great web framework. When I try to run my "asynchronous" app on Flask, I get that the session is attached to a different loop than the one I use to tear it down. Moreover, the sessions are piling up on on my registry which means that when Flask(or most likely me) is not tearing down the context properly.

The Error I get is:

```python
RuntimeError: Task <Task pending name='Task-8' coro=<AsyncToSync.main_wrap() running at /usr/local/lib/python3.11/site-packages/asgiref/sync.py:353> cb=[_run_until_complete_cb() at /usr/local/lib/python3.11/asyncio/base_events.py:180]> got Future <Future pending cb=[Protocol._on_waiter_completed()]> attached to a different loop
```

I added a reproducible example using devcontainer at: https://github.com/EM51641/async_flask_sqlalchemy_app/tree/main. Could somebody please tell me what am I doing so wrong ?

**Comment by nioncode** [comment]
@EM51641 did you find a solution to this?

**Reply by EM51641** [reply]
Hey @nioncode, unfortunately I didn't find one but switched to Quart and everything worked properly.
